pisa/stages/osc/prob3.py

```diff
@@ -398,32 +395,23 @@ def compute_function(self):
-                for container in self.data:
-                    self.layers.calcLayers(container['true_coszen'])
-                    container['densities'] = self.layers.density.reshape((container.size, self.layers.max_layers))
-                    container['distances'] = self.layers.distance.reshape((container.size, self.layers.max_layers))
-        elif self.tomography_type == "mass_of_core_wo_constrain":
-                self.tomography_params.core_density_scale = self.params.core_density_scale.value.m_as('dimensionless')
-                self.tomography_params.innermantle_density_scale = self.params.innermantle_density_scale.value.m_as('dimensionless')
-                self.tomography_params.middlemantle_density_scale = self.params.middlemantle_density_scale.value.m_as('dimensionless')
-                self.layers = Layers(earth_model, detector_depth, prop_height,scaling_array=self.tomography_params.scaling_factor_array)
-                self.layers.setElecFrac(self.YeI, self.YeO, self.YeM)
-                for container in self.data:
-                    self.layers.calcLayers(container['true_coszen'])
-                    container['densities'] = self.layers.density.reshape((container.size, self.layers.max_layers))
-                    container['distances'] = self.layers.distance.reshape((container.size, self.layers.max_layers))
+        if self.tomography_type is not None:
+            if self.tomography_type == "mass_of_earth":
+                    self.tomography_params.density_scale = self.params.density_scale.value.m_as('dimensionless')
+                    self.layers.scaling(scaling_array=self.tomography_params.density_scale)
+            elif self.tomography_type == "mass_of_core_w_constrain":
+                    self.tomography_params.core_density_scale = self.params.core_density_scale.value.m_as('dimensionless')
+                    self.layers.scaling(scaling_array=self.tomography_params.scaling_array)
+            elif self.tomography_type == "mass_of_core_wo_constrain":
+                    self.tomography_params.core_density_scale = self.params.core_density_scale.value.m_as('dimensionless')
+                    self.tomography_params.innermantle_density_scale = self.params.innermantle_density_scale.value.m_as('dimensionless')
+                    self.tomography_params.middlemantle_density_scale = self.params.middlemantle_density_scale.value.m_as('dimensionless')
+                    self.layers.scaling(scaling_array=self.tomography_params.scaling_factor_array)
+            self.layers.setElecFrac(self.YeI, self.YeO, self.YeM)        
+            for container in self.data:
+                self.layers.calcLayers(container['true_coszen'])
+                container['densities'] = self.layers.density.reshape((container.size, self.layers.max_layers))
+                container['distances'] = self.layers.distance.reshape((container.size, self.layers.max_layers))
```

Comment:
Do you also have to overwrite this? The scaling shouldn't change the distances, right?

Reply:
Yes right. We don't need to overwrite container['distances']. I have removed it. We only require till container['densities'].